implementation-notes.md

```diff
@@ -0,0 +1,94 @@
+A collection of Implementation notes

```

Comment:
Should you link to these notes from the spec?

Reply:
The plan is to do that yes when they are well organized. I removed parts from the spec and added it here to not lose the valuable info